LiteCore/Database/TreeDocument.cc

```diff
@@ -640,14 +644,51 @@ namespace litecore {
             return true;
         }
 
+        static bool hasEncryptables(slice body, SharedKeys* sk) {
+#ifndef COUCHBASE_ENTERPRISE
+            return false;
+#else
+            const Value* v = Value::fromTrustedData(body);
+            if (v == nullptr) {
+                return false;
+            }
+
+            struct AuxDoc : public Doc {
+                AuxDoc(slice body, SharedKeys* sk)
+                : Doc(alloc_slice(body), Doc::kTrusted, sk)
+                {}
+            };
+            AuxDoc doc(body, sk);
+
+            bool ret = false;
+            for (DeepIterator i(doc.asDict()); i; ++i) {
+                const Dict* dict = i.value()->asDict();
+                if (dict) {
+                    const Value* objType = dict->get(C4Document::kObjectTypeProperty);
+                    if (objType && objType->asString() == C4Document::kObjectType_Encryptable) {
+                        ret = true;
+                        break;
+                    }
+                }
+            }
+            return ret;
+#endif
+        }
 
-        static revidBuffer generateDocRevID(slice body, slice parentRevID, bool deleted) {
+        static revidBuffer generateDocRevID(slice body, slice parentRevID, bool deleted, SharedKeys* sk) {
             // Get SHA-1 digest of (length-prefixed) parent rev ID, deletion flag, and revision body:
             uint8_t revLen = (uint8_t)min((unsigned long)parentRevID.size, 255ul);
             uint8_t delByte = deleted;
-            SHA1 digest = (SHA1Builder() << revLen << slice(parentRevID.buf, revLen)
-                                         << delByte << body)
-                           .finish();
+            SHA1 digest;
+            if (hasEncryptables(body, sk)) {
```

Comment:
CBL can provide this info similar to kRevHasAttachments. Want to discuss if we would like CBL to provide or as this is a security issue, we want to do it in LiteCore to avoid bugs from CBL side?